content/blog/2023/07/22/2023-07-22-gsoc-2023-midterm.adoc

```diff
@@ -0,0 +1,188 @@
+
+Sharing details about the probes that I worked on during the first phase of GSoC.
+
+=== Unreleased Production Changes Probe
+
+A bug fix, security improvement, or new feature can only be beneficial when it's actually released.
+The "Unreleased Production Changes" probe is designed to detect unreleased commits in the production files.
+
+This probe's primary goal is to identify unreleased production commits specifically in the Maven production files.
+
+Following my mentors' advice, I began by writing test cases to accurately reproduce the probe's requirements. Subsequently, I added the necessary code to pass each of these test cases.
+
+During the implementation, I utilized the JGit library and familiarized myself with its features.
+However, I encountered some challenges due to a lack of depth in Git concepts, which required me to delve deeper and strengthen my understanding of Git concepts along the way.
+
+=== The Third-Party Repository Detection Probe
+
+By definition, a third-party repository is a repository that is not hosted at link:https://repo.jenkins-ci.org/[https://repo.jenkins-ci.org].
+
+The third-party repository detection probe ensures that plugins are only built from trusted and reliable Maven repositories.
+
+This probe was requested by the Jenkins community.
+
+One of the major challenges in implementing this probe was considering the parent and child pom relationships from all angles.
+
+Certain edge-case scenarios were not initially taken into account while specifying the probe, and due to the number of open questions, the probe was temporarily put on hold until further research was conducted.
+
+Throughout the process, I extensively read the Maven documentation to gain a better understanding of how to test POM structure, inheritance, and Maven API.
+
+Read the link:blog/2023/07/20/third-party-repository-detection-probe.adoc[unreleased production changes] blog post for more details on the probe.
+
+=== GitHub Security Scan Workflow Probe
+
+The Security Scan probe identifies whether the plugin is configured to execute the security action.
+
+From a security perspective, this probe is crucial as it checks for known security breaches and vulnerabilities in a plugin.
+
+Its primary purpose is to ensure that the security scan is correctly configured in the plugin's GitHub Action.
+
+Additionally, this probe involves refactoring existing classes to avoid code duplication, and it shares similarities with the JEP-229 (Continuous Delivery) probe, which also looks for CI configurations in GitHub Actions.
+
+Throughout this process, I have gained knowledge in leveraging the class hierarchy to build an abstract probe that can be implemented by any probes requiring GitHub Action configuration checks.
+
+The findings of this probe will be highly beneficial to the security team, as it helps identify any gaps and enables the improvement of the overall security of the Jenkins plugin ecosystem.
+
```

Comment:
Read the link:blog/2023/07/16/security-scan-probe-blog/.adoc[security scan probe] blog post for more details on the probe.

Reply:
Updated